tests/output_filters/run_filter_test.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+from contextlib import nested
+from StringIO import StringIO
+
+import simplejson
+import testify as T
+from mock import patch
+from osxcollector.output_filters.base_filters.output_filter import run_filter
+
+
+class RunFilterTest(T.TestCase):
+
+    @T.setup_teardown
+    def never_write_api_cache(self):
+        with patch('osxcollector.output_filters.util.api_cache.ApiCache._write_cache_to_file'):
+            yield
+
+    def run_test(self, create_filter, input_blobs=None, expected_output_blobs=None):
+        """Mocks out stdin, stdout, and config then runs input lines through an OutputFilter.
+
+        Args:
+            create_filter: A callable that returns an OutputFilter.
+            input_blobs: An array of dicts to pass to OutputFilter. These will be serialized into strings and passed as stdin.
+            expected_output_blobs: An array of dicts the output of the OutputFilter must match.
+            config_initial_contents: A dict of config.
+            api_cache_initial_contents: A dict of config.
+            blacklist_file_contents: An enumerable with contents of a blacklist file.
+        """
+        if not input_blobs:
+            input_blobs = []
+        input_lines = '\n'.join([simplejson.dumps(blob) for blob in input_blobs])
+
+        with nested(
+            patch('sys.stdin', StringIO(input_lines)),
+            patch('sys.stdout', new_callable=StringIO),
+            patch('osxcollector.output_filters.util.config._config_file_path',
+                  return_value='./tests/output_filters/data/test_osxcollector_config.yaml')
+        ) as (
+            mock_stdin,
+            mock_stdout,
+            __
+        ):
+            output_filter = create_filter()
+            run_filter(output_filter)
+            output_lines = [line for line in mock_stdout.getvalue().split('\n') if len(line)]
+            output_blobs = [simplejson.loads(line) for line in output_lines]
+
+            if expected_output_blobs:
+                T.assert_equal(len(output_blobs), len(expected_output_blobs))
+
+                for expected_blob, actual_blob in zip(expected_output_blobs, output_blobs):
+                    assert_equal_sorted(expected_blob, actual_blob)
+
+            return output_blobs
+
+    def assert_key_added_to_blob(self, added_key, expected_values, input_blobs, output_blobs):
+        """Verifies that a single key has been added to each input_blob with an expected value.
+
+        Asserts that effectively:
+        output_blobs = [input_blob.update(key=expected_value) for input_blob, expected_value in zip(expected_values, input_blobs)]
```

Comment:
...input_blob, expected_value in zip(expected_values, input_blobs)... should be:

expected_value, input_blob in zip(expected_values, input_blobs)
OR vice versa